app/src/main/java/com/lagradost/cloudstream3/ui/result/ResultViewModel2.kt

```diff
@@ -841,71 +904,243 @@ class ResultViewModel2 : ViewModel() {
         val response = currentResponse ?: return null
         if (response !is EpisodeResponse) return null
 
         val currentId = response.getId()
 
         if (isSubscribed) {
-            DataStoreHelper.removeSubscribedData(currentId)
+            removeSubscribedData(currentId)
+            statusChangedCallback?.invoke(false)
+            _subscribeStatus.postValue(false)
+            return false
         } else {
-            val current = DataStoreHelper.getSubscribedData(currentId)
+            checkAndWarnDuplicates(
+                context,
+                LibraryListType.SUBSCRIPTIONS,
+                CheckDuplicateData(
+                    name = response.name,
+                    year = response.year,
+                    syncData = response.syncData,
+                ),
+                getAllSubscriptions(),
+            ) { shouldContinue: Boolean, duplicateIds: List<Int?> ->
+                if (!shouldContinue) {
+                    statusChangedCallback?.invoke(null)
+                    return@checkAndWarnDuplicates
+                }
 
-            DataStoreHelper.setSubscribedData(
-                currentId,
-                DataStoreHelper.SubscribedData(
+                if (duplicateIds.isNotEmpty()) {
+                    duplicateIds.forEach { duplicateId ->
+                        removeSubscribedData(duplicateId)
+                    }
+                }
+
+                val current = getSubscribedData(currentId)
+
+                setSubscribedData(
                     currentId,
-                    current?.bookmarkedTime ?: unixTimeMS,
-                    unixTimeMS,
-                    response.getLatestEpisodes(),
-                    response.name,
-                    response.url,
-                    response.apiName,
-                    response.type,
-                    response.posterUrl,
-                    response.year
+                    DataStoreHelper.SubscribedData(
+                        current?.subscribedTime ?: unixTimeMS,
+                        response.getLatestEpisodes(),
+                        currentId,
+                        unixTimeMS,
+                        response.name,
+                        response.url,
+                        response.apiName,
+                        response.type,
+                        response.posterUrl,
+                        response.year,
+                        response.syncData
+                    )
                 )
-            )
-        }
 
-        _subscribeStatus.postValue(!isSubscribed)
-        return !isSubscribed
+                _subscribeStatus.postValue(true)
+
+                statusChangedCallback?.invoke(true)
+            }
+
+            return _subscribeStatus.value
+        }
     }
 
     /**
      * @return true if added to favorites, false if not. Null if not possible to favorite.
      **/
-    fun toggleFavoriteStatus(): Boolean? {
+    fun toggleFavoriteStatus(
```

Comment:
this function should not return anything as it uses the callback

Reply:
it goes without saying that the comment needs updating as well

Reply:
Thanks! Done.